src/plugins/discover/public/application/embeddable/saved_search_embeddable.tsx

```diff
@@ -0,0 +1,409 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Subscription } from 'rxjs';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { i18n } from '@kbn/i18n';
+import { isEqual } from 'lodash';
+import { Container, Embeddable } from '../../../../embeddable/public';
+import { ISearchEmbeddable, SearchInput, SearchOutput } from './types';
+import { SavedSearch } from '../../saved_searches';
+import { Adapters, RequestAdapter } from '../../../../inspector/common';
+import { SEARCH_EMBEDDABLE_TYPE } from './constants';
+import { APPLY_FILTER_TRIGGER, esFilters, FilterManager } from '../../../../data/public';
+import { DiscoverServices } from '../../build_services';
+import {
+  Query,
+  TimeRange,
+  Filter,
+  IFieldType,
+  IndexPattern,
+  ISearchSource,
+} from '../../../../data/common';
+import { SortOrder } from '../angular/doc_table/components/table_header/helpers';
+import { ElasticSearchHit } from '../doc_views/doc_views_types';
+import { SavedSearchEmbeddableComponent } from './saved_search_embeddable_component';
+import { UiActionsStart } from '../../../../ui_actions/public';
+import { getServices } from '../../kibana_services';
+import {
+  DOC_HIDE_TIME_COLUMN_SETTING,
+  SAMPLE_SIZE_SETTING,
+  SEARCH_FIELDS_FROM_SOURCE,
+  SORT_DEFAULT_ORDER_SETTING,
+} from '../../../common';
+import * as columnActions from '../angular/doc_table/actions/columns';
+import { getSortForSearchSource, getDefaultSort } from '../angular/doc_table';
+import { handleSourceColumnState } from '../angular/helpers';
+import { DiscoverGridProps } from '../components/discover_grid/discover_grid';
+import { DiscoverGridSettings } from '../components/discover_grid/types';
+
+export interface SearchProps extends Partial<DiscoverGridProps> {
+  settings?: DiscoverGridSettings;
+  description?: string;
+  sharedItemTitle?: string;
+  inspectorAdapters?: Adapters;
+
+  filter?: (field: IFieldType, value: string[], operator: string) => void;
+  hits?: ElasticSearchHit[];
+  totalHitCount?: number;
+  onMoveColumn?: (column: string, index: number) => void;
+  useLegacyTable?: boolean;
+  refs?: HTMLElement;
+}
+
+interface SearchEmbeddableConfig {
+  savedSearch: SavedSearch;
+  editUrl: string;
+  editPath: string;
+  indexPatterns?: IndexPattern[];
+  editable: boolean;
+  filterManager: FilterManager;
+  services: DiscoverServices;
+}
+
```

Comment:
I'd suggest using a memoized version of SavedSearchEmbeddableComponent the component SavedSearchEmbeddableComponent
```
const SavedSearchEmbeddableComponentMemoized = React.memo(SavedSearchEmbeddableComponent);
```
this reduces the rendering count when loading the embeddable to 2 (from 3)

<img width="2551" alt="Bildschirmfoto 2021-06-07 um 19 22 03" src="https://user-images.githubusercontent.com/463851/121062728-c145e480-c7c5-11eb-833a-f65fd60df8d3.png">